src/node/internal/internal_assert.ts

```diff
@@ -934,9 +934,9 @@ function validateThrownError(
   if (
     error instanceof Function &&
     error.prototype !== undefined &&
-    error.prototype instanceof Error
+    (error === Error || error.prototype instanceof Error)
   ) {
-    // error is a constructor
```

Comment:
Curious... does this actually impact the web platform tests? If so, it's not obvious and could use an additional comment in the PR. Otherwise, this just feels like it's a different issue than the TextDecoderStream fix. Either way, this change could likely better go into a separate commit; non-blocking tho.

Reply:
Got it. Ok, for the sake of others, the custom test harness that we use to run the WPT's depends on `node:assert` and this specific change. The connection there is non-obvious in this PR so just clarifying.